lib/runner_test.go

```diff
@@ -373,155 +378,265 @@ func TestShuffleOrder(t *testing.T) {
+				t.Fatalf("Expected stdout:\n %q\n\nHave:\n %q\n", tt.expectedStdout, stdoutStr)
+			}
+			stderrBytes, err := ioutil.ReadAll(stderr)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if stderrStr := string(stderrBytes); stderrStr != tt.expectedStderr {
+				t.Fatalf("Expected stderr:\n %q\n\nHave:\n %q\n", tt.expectedStderr, stderrStr)
+			}
+		})
 	}
 }
 
 func TestOutputResults(t *testing.T) {
-	_, out := setupTestUI()
-	r := setupDefaultRunner()
-	r.RandomSeed = 42
+	r, stdout, stderr := setupDefaultRunner()
+	r.options.randomSeed = 42
 	r.testResults = setupTestResults()
 	r.failedTestResults = setupFailedTestResults()
 
 	r.outputResults()
-	expected := redBoldString("testfile-failure-1: Failed with exit code 1\n") +
+	expectedStdout := "Seed used: 42\n"
+	expectedStderr := redBoldString("testfile-failure-1: Failed with exit code 1\n") +
 		redBoldString("testfile-failure-2: Failed with exit code 2\n") +
-		redBoldString("\nTests complete: 2 Passed, 2 Failed\n") +
-		"Seed used: 42\n"
-	if got := out.String(); got != expected {
-		t.Fatalf("Expected:\n %q\n\nHave:\n %q\n", expected, got)
+		redBoldString("\nTests complete: 2 Passed, 2 Failed\n")
+	stdoutBytes, err := ioutil.ReadAll(stdout)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if stdoutStr := string(stdoutBytes); stdoutStr != expectedStdout {
+		t.Fatalf("Expected stdout:\n %q\n\nHave:\n %q\n", expectedStdout, stdoutStr)
+	}
+	stderrBytes, err := ioutil.ReadAll(stderr)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if stderrStr := string(stderrBytes); stderrStr != expectedStderr {
+		t.Fatalf("Expected stderr:\n %q\n\nHave:\n %q\n", expectedStderr, stderrStr)
 	}
 }
 
 func TestOutputResultsJSON(t *testing.T) {
-	_, out := setupTestUI()
-	r := setupDefaultRunner()
+	r, stdout, stderr := setupDefaultRunner()
 	r.failedTestResults = setupFailedTestResults()
-	r.RandomSeed = 42
+	r.options.randomSeed = 42
 	r.testResults = setupTestResults()
 	r.failedTestResults = setupFailedTestResults()
 
 	r.outputResultsJSON()
-	expected := `{"passed":2,"failed":2,"seed":42,"inOrder":false,"failedList":[` +
-		`{"filename":"testfile-failure-1","success":false,"exitcode":1,"output":"It didn't work!"},` +
-		`{"filename":"testfile-failure-2","success":false,"exitcode":2,"output":"It didn't work again!"}]}`
-	if got := out.String(); got != expected {
-		t.Fatalf("Expected:\n %q\n\nHave:\n %q\n", expected, got)
+	expectedStdout := `{"passed":2,"failed":2,"seed":42,"inOrder":false,"failedList":[` +
+		`{"filename":"testfile-failure-1","success":false,"exitcode":1},` +
+		`{"filename":"testfile-failure-2","success":false,"exitcode":2}]}` +
+		"\n"
+	expectedStderr := ""
+	stdoutBytes, err := ioutil.ReadAll(stdout)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if stdoutStr := string(stdoutBytes); stdoutStr != expectedStdout {
+		t.Fatalf("Expected stdout:\n %q\n\nHave:\n %q\n", expectedStdout, stdoutStr)
+	}
+	stderrBytes, err := ioutil.ReadAll(stderr)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
Noticed all the tests are doing `t.Fatal` and not `t.Error`, are you not intending for the test to continue after the first failure? I would generally use fatal if I want to do more debugging in the failing test and you care about the stack trace.